scripts/run.sh

```diff
@@ -113,43 +114,64 @@ INPUT_DIR=${BASEDIR}/EVGEN/${TAG}
 mkdir -p ${INPUT_DIR}
 TAG=${DETECTOR_VERSION:-main}/${DETECTOR_CONFIG}/${TAG}
 
-# Define location on xrootd from where to stream input file from
-INPUT_FILE=${XRDRURL}/${XRDRBASE}/${INPUT_FILE}
+if [[ "$EXTENSION" == "hepmc3.tree.root" ]]; then
+  # Define location on xrootd from where to stream input file from
+  INPUT_FILE=${XRDRURL}/${XRDRBASE}/${INPUT_FILE}
+else
+  # Copy input file from xrootd
+  xrdcp -f ${XRDRURL}/${XRDRBASE}/${INPUT_FILE} ${INPUT_DIR}
+fi
 
 # Output file names
 LOG_DIR=LOG/${TAG}
 LOG_TEMP=${TMPDIR}/${LOG_DIR}
-mkdir -p ${LOG_TEMP} ${BASEDIR}/${LOG_DIR}
+mkdir -p ${LOG_TEMP} 
 #
 FULL_DIR=FULL/${TAG}
 FULL_TEMP=${TMPDIR}/${FULL_DIR}
-mkdir -p ${FULL_TEMP} ${BASEDIR}/${FULL_DIR}
+mkdir -p ${FULL_TEMP} 
 #
 RECO_DIR=RECO/${TAG}
 RECO_TEMP=${TMPDIR}/${RECO_DIR}
-mkdir -p ${RECO_TEMP} ${BASEDIR}/${RECO_DIR}
+mkdir -p ${RECO_TEMP} 
 
 # Run simulation
 {
   date
   eic-info
+  # Common flags shared by both types of simulation
+  common_flags=(
+    --random.seed ${SEED:-1}
+    --random.enableEventSeed
+    --printLevel WARNING
+    --filter.tracker 'edep0'
+    --numberOfEvents ${EVENTS_PER_TASK}
+    --outputFile ${FULL_TEMP}/${TASKNAME}.edm4hep.root
+  )
+  # Uncommon flags based on EXTENSION
+  if [[ "$EXTENSION" == "hepmc3.tree.root" ]]; then
+    uncommon_flags=(
+      --runType batch
+      --skipNEvents ${SKIP_N_EVENTS}
+      --hepmc3.useHepMC3 ${USEHEPMC3:-true}
+      --compactFile ${DETECTOR_PATH}/${DETECTOR_CONFIG}${EBEAM:+${PBEAM:+_${EBEAM}x${PBEAM}}}.xml
+      --inputFiles ${INPUT_FILE}
+    )
+  else
+    uncommon_flags=(
+      --runType run
```

Comment:
Is this a necessary difference?

Reply:
I had it initially for the singles to always default to the 5x41 field even if the EBEAM and PBEAM environment variable was set to something else. More to prevent human error in calling with a different field setting from one campaign to another.

Reply:
Sorry, it defaults to 18x275 actually.

Reply:
![Uploading Screenshot_20241219_090155_Chrome.jpg…]()

Reply:
https://github.com/AIDASoft/DD4hep/blob/abb85cbacd3909ea54733f2defd8053eaee63d12/DDG4/python/DDSim/DD4hepSimulation.py#L201

Reply:
May be that runType="run" is necessary when using Particle Gun